### PR TITLE
Milestone 6: Add views to the blog app

### PR DESCRIPTION
**Pull request by Kandy-Peter**
- [ ] Implement the design from the sneak peek wireframes.

- [ ] Use methods created in the [Blog App - processing data in models](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md).
- [ ] Add some data to your database as described [here](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md#use-models-to-insert-data).
- [ ] Fix linters errors

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1 +1,15 @@
...
+          <img src="<%= user.photo %>" alt="user_pic" />
+        </li>
+        <li>
+          <%= user.name %>
+          <p class="posts_count">Number of posts: <%= user.posts_counter || 0 %></p>
+        </li>
+      </ul>
+    </li>
+  <% end %>
+</ul>
```
Comment (andersonlebon) [inline app/views/users/index.html.erb]: image


- [ ] Kindly note that you need to follow the wireframes that were provided in the requirements, I know that the style does not matter but you need to add some basic CSS to follow the [wireframes](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Views_project.md). check the wireframes :point_down: 

image

- [ ] Kindly also make sure when I click on the user it redirects me to the `user#show` view page

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1 +1,26 @@
...
+        <p>User doesn't have posts yet</p>
+      <% else %>
+        <div>
+            <h2><%= post.title %></h2>
+            <p><%= post.text %></p>
+            <h3>Comment <%= post.comments.count%>,  <span>Likes <%= post.likes.count%></span></h3>
+        </div>
+      <% end %>
+    <% end %>
+  </ul>
```
Comment (andersonlebon) [inline app/views/users/show.html.erb]: image



- [ ] Kindly note that you need to follow the wireframes that were provided in the requirements, I know that the style does not matter but you need to add some basic CSS to follow the [wireframes](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Views_project.md). check the wireframes :point_down: 

image

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -1 +1,35 @@
...
+          <% else %>
+            <% post.most_recent_comments.each do |comment| %>
+              <%= comment.author.name %>  :<%= comment.text %>
+            <% end %>
+          <% end %>
+        </div>
+      <% end %>
+    <% end %>
+  </div>
+</div>
```
Comment (andersonlebon) [inline app/views/posts/index.html.erb]: image


- [ ] Kindly note that you need to follow the wireframes that were provided in the requirements, I know that the style does not matter but you need to add some basic CSS to follow the [wireframes](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Views_project.md). check the wireframes :point_down: 

image

- [ ] Kindly also make sure when I click on the post, it redirects me to the `post#show` view page.

File: app/views/posts/show.html.erb [hunk 4]
```diff
@@ -1 +1,21 @@
...
+  </div>
+  <div>
+    <% if @post.most_recent_comments.empty? %>
+        <p>No comment for this post</p>
+    <% else %>
+      <% @post.most_recent_comments.each do |comment| %>
+        <%= comment.author.name %>  :<%= comment.text %>
+      <% end %>
+    <% end %>
+  </div>
```
Comment (andersonlebon) [inline app/views/posts/show.html.erb]: image


- [ ] Kindly note that you need to follow the wireframes that were provided in the requirements, I know that the style does not matter but you need to add some basic CSS to follow the [wireframes](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Views_project.md). check the wireframes :point_down: 

image

File: app/controllers/posts_controller.rb [hunk 5]
```diff
@@ -1,5 +1,10 @@
 class PostsController < ApplicationController
-  def index; end
+  def index
+    @user = User.find_by(id: current_user)
+    @posts = Post.where(author_id: current_user)
+  end
```
Comment (andersonlebon) [inline app/controllers/posts_controller.rb:6]: ## [OPTIONAL]

image

## 1. 
int your code mode `models/post.rb:line 8...`


```rb
  def update_post_counter
    count = Post.where(author_id: id).count
    user = User.find_by_id(id: author_id)
    user.update(posts_counter: count)
  end
```

- [ ] I would really suggest you solve this error when the post is created. you can solve it by improving the method `update_post_counter` with

```rb
  def update_post_counter
    author.increment!(:posts_counter)
  end
```



## 2
int your code mode `models/comment.rb:line 7...`

```rb
  def update_comment_counter
    count = Comment.where(post_id: id).count
    post = Post.find_by_id(id: post_id)
    post.update(comments_counter: count)
  end
```

- [ ] The same error is in the comment modal also, kindly solve it.  you can solve it with
```rb 
  def update_comment_counter
    post.increment!(:comments_counter)
  end
```
image

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @Kandy-Peter 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
## Required Changes ♻️

- [ ] Kindly check comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 5 above)
Comment (Kandy-Peter) [reply app/controllers/posts_controller.rb:6]: Hi @andersonlebon,

Thank you for your suggestions. I really appreciate them.

File: app/views/users/index.html.erb (hunk 1 above)
Comment (DeliceLydia) [reply app/views/users/index.html.erb]: Addressed

**Review by DeliceLydia — APPROVED** [review]
## Hi @Kandy-Peter, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**